pkg/managed/serviceaccount_test.go

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) 2020, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package managed
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/verrazzano/verrazzano-operator/pkg/monitoring"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/verrazzano/verrazzano-operator/pkg/constants"
+	"github.com/verrazzano/verrazzano-operator/pkg/testutil"
+	"github.com/verrazzano/verrazzano-operator/pkg/types"
+	"github.com/verrazzano/verrazzano-operator/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8slabels "k8s.io/apimachinery/pkg/labels"
+)
+
+// TestCreateAppServiceAccounts tests creation of service accounts when an application binding is used.
+// GIVEN clusters which do not have service accounts for each namespace of an application binding
+//  WHEN I call CreateServiceAccounts
+//  THEN there should be a service account created for each namespace of an application binding
+//   AND that service account should have the expected values
+func TestCreateAppServiceAccounts(t *testing.T) {
+	// Setup the needed structures to test creating service accounts for an application binding.
+	imagePullSecrets := []corev1.LocalObjectReference{
+		{
+			Name: "test-imagePullSecret",
+		},
+	}
+	managedClusters := getManagedClusters()
+	managedConnections := getManagedConnections()
+	createAppNamespaces(t, managedConnections)
+
+	// Create the expected service accounts
+	err := CreateServiceAccounts("testBinding", imagePullSecrets, managedClusters, managedConnections)
+	if err != nil {
+		t.Fatal(fmt.Sprintf("can't create service account: %v", err))
+	}
+
+	// Validate the service accounts created for the "local" cluster
+	managedClusterConnection := managedConnections["local"]
+	serviceAccounts, err := managedClusterConnection.ServiceAccountLister.List(k8slabels.Everything())
+	if err != nil {
+		t.Fatal(fmt.Sprintf("can't list service account for cluster local: %v", err))
+	}
+	assert.Equal(t, 2, len(serviceAccounts), "2 service accounts were expected for cluster local")
```

Comment:
could part of this validation make sure that the names of the two service accounts created are the expected one?

Reply:
Yes, the names of the expected service accounts are checked in the assertCreate... functions